party/party.go

```diff
@@ -109,3 +110,33 @@ func (p *Party) SetOwner(userUUID UserUUID) error {
 
 	return nil
 }
+
+// updated increments the update tracker.
+// Should call this whenever there's an update everyone should know about.
+func (p *Party) setUpdated() {
+	p.changeID++
+}
+
+// Pull returns the user data in a serializable format.
+// NOTE: this checks for changes before checking uid.
+func (p *Party) Pull(userUUID UserUUID, changeID uint64) (interface{}, error) {
+	p.mux.Lock()
+	defer p.mux.Unlock()
+
+	// TODO: what if change ID is larger?
```

Comment:
Or return nil, nil again. The latter is probably easier. It's the client's fault if it's an idiot.